src/main/java/kaim/command/Parser.java

```diff
@@ -0,0 +1,188 @@
+     *
+     * @param parts The command parts (description) for the todo task.
+     * @param tasks The current task list to add the task to.
+     * @return A response message indicating that the task has been added.
+     * @throws KaiMException If the description of the todo is empty.
+     */
+    private String addTodo(String[] parts, TaskList tasks) throws KaiMException {
+        if (parts.length < 2) {
+            throw new KaiMException("The description of a todo cannot be empty.");
+        }
+        Task task = new Todo(parts[1].trim());
+        tasks.addTask(task);
+        return "Added: " + task;
+    }
+
+    /**
+     * Adds a new Deadline task to the task list.
+     *
+     * @param parts The command parts (description and deadline) for the task.
+     * @param tasks The current task list to add the task to.
+     * @return A response message indicating that the task has been added.
+     * @throws KaiMException If the deadline format is incorrect.
+     */
+    private String addDeadline(String[] parts, TaskList tasks) throws KaiMException {
+        String[] details = parts[1].split("/by");
+        if (details.length < 2) {
+            throw new KaiMException("Deadline format: deadline task /by yyyy-MM-dd HHmm");
+        }
+        Task task = new Deadline(details[0].trim(), details[1].trim());
+        tasks.addTask(task);
+        return "Added: " + task;
+    }
+
+    /**
+     * Adds a new Event task to the task list.
+     *
+     * @param parts The command parts (description, from, and to times) for the event task.
+     * @param tasks The current task list to add the task to.
+     * @return A response message indicating that the task has been added.
+     * @throws KaiMException If the event format is incorrect.
+     */
+    private String addEvent(String[] parts, TaskList tasks) throws KaiMException {
+        String[] details = parts[1].split("/from|/to");
+        if (details.length < 3) {
+            throw new KaiMException("Event format: event task /from start /to end");
+        }
+        Task task = new Event(details[0].trim(), details[1].trim(), details[2].trim());
+        tasks.addTask(task);
+        return "Added: " + task;
+    }
+
+    /**
+     * Marks or unmarks a task as done or not done.
+     *
+     * @param parts The command parts (task number) to toggle the task status.
+     * @param tasks The current task list to modify.
+     * @param mark  True to mark the task as done, false to mark it as not done.
+     * @return A response message indicating the task status has been toggled.
+     * @throws KaiMException If the task index is invalid.
+     */
+    private String toggleTaskStatus(String[] parts, TaskList tasks, boolean mark) throws KaiMException {
+        int index = Integer.parseInt(parts[1]) - 1;
+        tasks.markTask(index, mark);
+        return mark ? "Marked as done: " + tasks.getTask(index) : "Marked as not done: " + tasks.getTask(index);
+    }
```

Comment:
I thought it may be better if catching the exception with some user-friendly error message
```suggestion
    private String toggleTaskStatus(String[] parts, TaskList tasks, boolean mark) throws KaiMException {
        try {
            int index = Integer.parseInt(parts[1]) - 1;
            tasks.markTask(index, mark);
            return mark 
                            ? "Marked as done: " + tasks.getTask(index) 
                            : "Marked as not done: " + tasks.getTask(index);
        } catch (KaiMException) {
            throw new KaiMException("Invalid input. Task number must be a number.");
        }
```